firedrake/slate/slac/kernel_builder.py

```diff
@@ -577,17 +577,20 @@ def collect_coefficients(self):
         are dicts of form {coeff: (name, extent)} and mixed coefficients are
         double dicts of form {mixed_coeff: {coeff_per_space: (name, extent)}}.
         """
-        coeffs = self.expression.coefficients()
+        coeffs = self.expression.slate_coefficients()  # Here we need the slate coefficients so that we can access the splits a few lines down
         coeff_dict = OrderedDict()
-        for i, c in enumerate(coeffs):
-            element = c.ufl_element()
-            if type(element) == MixedElement:
-                mixed = OrderedDict()
-                for j, c_ in enumerate(c.split()):
-                    mixed[c_] = f"w_{i}_{j}", self.extent(c_)
-                coeff_dict[c] = mixed
+        for i, (c, split_map) in enumerate(self.expression.coeff_map):
+            coeff = coeffs[c]
+            splits = coeff.split_function if isinstance(coeff, slate.BlockFunction) else coeff.split() if type(coeff.ufl_element()) == MixedElement else ()
+            orig_coeff = coeff.orig_function if isinstance(coeff, slate.BlockFunction) else coeffs[c]
+            if splits:
+                mixed = OrderedDict() if coeff not in coeff_dict else coeff_dict[coeff]
+                for j in split_map:
+                    coeff_ = splits[j]
```

Comment:
`splits = coeff.split_function` if `coeff` is `BlockFunction`, but `split_map` is a tuple of indices into `coeff.orig_function.split()`, right?

Reply:
Ok, so I think `splits` here must be `coeff.orig_function.split()` instead of `coeff.split_function`; otherwise, if you replace indices `0` with `1` in `test_push_assembled_vector_blocks_individual`, this test fails.